controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,205 @@
+package storagecluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	rookCephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"reflect"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+type ConnectedClient struct {
+	Metadata  types.NamespacedName `json:"metadata"`
+	ClusterID string               `json:"clusterId"`
+}
+type InfoStorageCluster struct {
+	Metadata                types.NamespacedName `json:"metadata"`
+	StorageProviderEndpoint string               `json:"storageProviderEndpoint"`
+	CephClusterFSID         string               `json:"cephClusterFSID"`
+}
+
+type OdfInfoData struct {
```

Comment:
When doing camel-case names, it's common to treat acronyms more than two letters long as just another word, and often even two-letter acronyms. This is the stylistic precedent already set in the files being changed in this PR, and also I like it better, so it should be fine to leave as-is.

Reply:
I pretty much followed the preexisting naming convention. Another thought I had was that it's a bit unusual to name it `Odf...` instead of the original `Ocs...` as the other names (structs, operator, etc.) are still prefixed with the original.

Reply:
That is a long story about the rebranding of OCS that happened a couple of years ago.